profiles/killbill/src/test/java/org/killbill/billing/jaxrs/TestCache.java

```diff
@@ -0,0 +1,131 @@
+
+        // verify that they are not null and have the accountId stored as a key (the account created before)
+        Assert.assertNotNull(accountRecordIdCache);
+        Assert.assertNotNull(accountRecordIdCache.get(input.getAccountId().toString()));
+        Assert.assertNotNull(accountImmutableCache);
+        Assert.assertNotNull(accountImmutableCache.get(input.getAccountId()));
+        Assert.assertNotNull(accountBcdCache);
+        Assert.assertNotNull(accountBcdCache.get(input.getAccountId()));
+
+        // invalidate caches per account level by accountId
+        killBillClient.invalidateCacheByAccount(input.getAccountId().toString(), requestOptions);
+
+        // verify that now the caches don't have the accountId key stored
+        Assert.assertNull(accountRecordIdCache.get(input.getAccountId().toString()));
+        Assert.assertNull(accountImmutableCache.get(input.getAccountId()));
+        Assert.assertNull(accountBcdCache.get(input.getAccountId()));
+    }
+
+    @Test(groups = "slow", description = "Can Invalidate (clear) all Tenant Caches for current Tenant")
+    public void testInvalidateCacheByTenant() throws Exception {
+
+        createAccountWithPMBundleAndSubscriptionAndWaitForFirstInvoice();
+
+        // get all caches per tenant level
+        final Ehcache tenantRecordIdCache = cacheManager.getEhcache(CacheType.TENANT_RECORD_ID.getCacheName());
+        final Ehcache tenantPaymentStateMachineConfigCache = cacheManager.getEhcache(CacheType.TENANT_PAYMENT_STATE_MACHINE_CONFIG.getCacheName());
+        final Ehcache tenantCache = cacheManager.getEhcache(CacheType.TENANT.getCacheName());
+        final Ehcache tenantKvCache = cacheManager.getEhcache(CacheType.TENANT_KV.getCacheName());
+        final Ehcache tenantConfigCache = cacheManager.getEhcache(CacheType.TENANT_CONFIG.getCacheName());
+        final Ehcache tenantOverdueConfigCache = cacheManager.getEhcache(CacheType.TENANT_OVERDUE_CONFIG.getCacheName());
+        final Ehcache tenantCatalogCache = cacheManager.getEhcache(CacheType.TENANT_CATALOG.getCacheName());
+
+        DefaultTenant currentTenant = (DefaultTenant) tenantCache.get(tenantCache.getKeys().get(0)).getObjectValue();
+
+        // verify that they are not null and have the expected tenant information
+        Assert.assertNotNull(tenantRecordIdCache);
+        Assert.assertNotNull(tenantRecordIdCache.get(currentTenant.getId().toString()));
+        Assert.assertNotNull(tenantPaymentStateMachineConfigCache);
+        String tenantPaymentStateMachineConfigCacheKey = "PLUGIN_PAYMENT_STATE_MACHINE_noop::1";
+        Assert.assertNotNull(tenantPaymentStateMachineConfigCache.get(tenantPaymentStateMachineConfigCacheKey));
+        Assert.assertNotNull(tenantCache);
+        Assert.assertNotNull(tenantCache.get(DEFAULT_API_KEY));
+        Assert.assertNotNull(tenantKvCache);
+        String tenantKvCacheKey = "PUSH_NOTIFICATION_CB::1";
```

Comment:
It doesn't look like you registered an URL specifically, so I'm wondering what this value is? Does the test work if run by itself (I want to make sure it's not a value left over from another test)?